### PR TITLE
WIP: Social links block.

### PR DESCRIPTION
**Pull request by v9ai**
## Description
Implements: https://github.com/WordPress/gutenberg/issues/1873

## References
[slack discussion](https://wordpress.slack.com/archives/C02QB2JS7/p1553901163077800)
[Prototype](https://www.figma.com/proto/gql7fzWm5Ch401zkrLnBgvqU/social-icons-block-i2?node-id=0%3A2040&scaling=min-zoom)
## Checklist:
- [ ] My code is tested.
- [ ] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate.

### CODE REVIEW

**Comment by Soean** [comment]
I haven't tested it in the editor yet, but we should follow #14770 and add a `block.json` file for new blocks.

**Comment by mapk** [comment]
@nicolad thanks for taking this on!

I tried testing this PR and couldn't get through it. Here's a screencast of my experience. I'm also aware this is probably a work in progress so feel free to ignore my comments right now.

social


**A couple design notes:**

1. The block icon should not be a social media site. Maybe try a "sharing" icon like: https://material.io/tools/icons/?icon=share&style=baseline

2. The `+` button next to the URL is a bit confusing for me. Does it _submit_ the URL I entered? or does it add a new line? I couldn't get it to work.

**Comment by nicolad** [comment]
> I haven't tested it in the editor yet, but we should follow #14770 and add a `block.json` file for new blocks.

Thank you for review, had added `block.json`.

**Comment by nicolad** [comment]
> @nicolad thanks for taking this on!
> 
> I tried testing this PR and couldn't get through it. Here's a screencast of my experience. I'm also aware this is probably a work in progress so feel free to ignore my comments right now.
> 
> social
> 
> **A couple design notes:**
> 
> 1. The block icon should not be a social media site. Maybe try a "sharing" icon like: https://material.io/tools/icons/?icon=share&style=baseline
> 2. The `+` button next to the URL is a bit confusing for me. Does it _submit_ the URL I entered? or does it add a new line? I couldn't get it to work.
Yes, it is in progress. Was a little busy and did not had the chance to make a lot of things here. Will have more time upcoming days and will try to do more.
Anyway, thanks for the feedback it's always great to have an additional pair of eyes.

**Comment by Soean** [comment]
>Thank you for review, had added block.json.

Looks good, `save` should also live in its own file, see #14882 for example.

File: packages/block-library/src/social/index.js [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+export { metadata, name };
+
+export const settings = {
+	title: __( 'Social Links' ),
+
+	description: __( 'Display a row of icons of your social media accounts.' ),
+
+	icon,
+
+	category: 'layout',
```
Comment (nicolad) [inline packages/block-library/src/social/index.js]: What will be the better category fit?

**Comment by nicolad** [comment]
Thanks @Soean, updated.
I started doing this block by copying `button` block.
Could you please suggest other suitable blocks at which I can take a look at the implementation?

File: packages/block-library/src/social/index.js (hunk 1 above)
Comment (Soean) [reply packages/block-library/src/social/index.js]: `widget`?

**Comment by Wingo5315** [comment]
I think the icon shouldn't be the Facebook logo, as people would think it is a Facebook-specific block. Instead, I recommend replacing it with a Share icon.

File: packages/block-library/src/social-links/edit.js [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+
+		return {
+			childLinks: block ? block.innerBlocks : DEFAULT_EMPTY_ARRAY,
+		};
+	} ),
+	withDispatch( ( dispatch, { clientId, childLinks } ) => {
+		return {
+			addLink() {
+				const created = createBlock( 'core/social-link', { verticalAlignment: 'top' } );
+				dispatch( 'core/editor' ).insertBlock( created, undefined, clientId );
```
Comment (nicolad) [inline packages/block-library/src/social-links/edit.js]: Hello @gziolo, on this action I want to create a sub-child, for some reason it won't fire the needed result.
I am not sure what I am doing wrong here.

Comment (nicolad) [reply packages/block-library/src/social-links/edit.js]: It seems that `insertBlock` isn't the best approach here, tried a different solution, based of `columns` block approach: https://github.com/WordPress/gutenberg/pull/14855/commits/792401a25c043c634de116ec7f1acf9701e9d95e

**Comment by karmatosed** [comment]
As this had needs design feedback, I thought I would give the current PR a go. I am not sure if something went wrong as the experience I don't think was what intended. For example on loading I saw this:

image

Then I get:

image

I'd be happy to test again but I do think either something is a little weird my end or the PR needs a little nurturing. If this is a 'me' issue, I would be happy to review based on screenshots of the flow.

**Comment by nicolad** [comment]
> As this had needs design feedback, I thought I would give the current PR a go. I am not sure if something went wrong as the experience I don't think was what intended. For example on loading I saw this:
> 
> image
> 
> Then I get:
> 
> image
> 
> I'd be happy to test again but I do think either something is a little weird my end or the PR needs a little nurturing. If this is a 'me' issue, I would be happy to review based on screenshots of the flow.

hey @karmatosed, you're right - this PR needs a little bit of nurture since it's still a work in progress. I will let you know when it's ready to be reviewed. Thanks.

**Comment by mkaz** [comment]
@nicolad here is a quick GIF demo of what I have currently in place. I put that together prior to seeing the prototype from #1873 

I have it as a separate plugin which is pretty rough with the interactions. I can put it up somewhere if we want iterate, but after looking at the prototype in that issue and your work, maybe I can combine the two efforts to some degree

social-icons-demo

**Comment by mkaz** [comment]
@nicolad if it is alright with you, I'm going to combine our efforts, though I'm not quite sure how.
I don't think I can push to your branch, and not sure if you can push to one I create here.

**Comment by nicolad** [comment]
Closed in favor of this PR: https://github.com/WordPress/gutenberg/pull/16897

**Pull request closed without merging**